docs/source/executing-operations/query-batching.mdx

```diff
@@ -0,0 +1,191 @@
+
+### Configure client
+
+To enable batching in a client, configure `BatchHttpLink`. For details on implementing `BatchHttpLink`, see [batching operations](https://www.apollographql.com/docs/react/api/link/apollo-link-batch-http/).
+
+### Configuration compatibility
+
+If the router receives a query batch from a client, and batching is *not* enabled, the router sends a `BATCHING_NOT_ENABLED` error to the client, which clearly identifies why the query batch failed.
+
+## Metrics for query batching
+
+Metrics in the Apollo Router for query batching:
+
+<table class="field-table metrics">
+  <thead>
+    <tr>
+      <th>Name</th>
+      <th>Attributes</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+
+<tbody>
+<tr class="required">
+<td style="min-width: 150px;">
+
+##### `apollo.router.operations.batching`
+
+</td>
+<td>
+
+mode
+
+</td>
+<td>
+
+Counter for the number received batch requests.
+
+</td>
+</tr>
+
+</tbody>
+</table>
+
+
+## Query batch formats
+
+### Request format
+
+```graphql
+[
+query MyFirstQuery {
+  me {
+    id
+  }
+},
+query MySecondQuery {
+  me {
+    name
+  }
+}
+]
+```
+
+### Response format
+
+Responses are provided in JSON array, with ordering of responses matching the supplied request.
+
+```json
+[
+  {"data":{"me":{"id":"1"}}},
+  {"data":{"me":{"name":"Ada Lovelace"}}}
+]
+```
+
+## Error handling for query batching
+
+### Batch error
+
+If a batch of queries cannot be processed, the entire batch will fail. 
+
+For example, this query is invalid because it has two commas to separate the batch of queries:
+
+```graphql
+[
+query MyFirstQuery {
+  me {
+    id
+  }
+},,
+query MySecondQuery {
+  me {
+    name
+  }
+}
+]
+```
+
+As a result, an error is returned for the invalid batch:
+
+```json
+{"errors":
+  [
+    {"message":"Invalid GraphQL request","extensions":{"details":"failed to deserialize the request body into JSON: expected value at line 1 column 54","code":"INVALID_GRAPHQL_REQUEST"}}
+  ]
+}
+```
+
+### Individual query error
+
+If a single query in a batch cannot be processed, this will result in an individual error. 
+
+For example, the query `MyFirstQuery` is accessing a field that doesn't exist, while the rest of the batch query is valid.
+
+```graphql
+[
+query MyFirstQuery {
+  me {
+    thisfielddoesnotexist
+  }
+},
+query MySecondQuery {
+  me {
+    name
+  }
+}
+]
+```
+
+As a result, an error is returned for the individual query:
+
+```json
+[
+  {"errors":
+    [
+      {"message":"cannot query field 'thisfielddoesnotexist' on type 'User'",
+       "extensions":{"type":"User","field":"thisfielddoesnotexist","code":"INVALID_FIELD"}
+      }
+    ]
+  },
+  {"data":{"me":{"name":"Ada Lovelace"}}}
+]
+```
+
+## Known limitations
+
+### Unsupported query modes
+ 
+When batching is enabled, any mode of querying that results in a stream of responses is unsupported, including:
```

Comment:
You mentioned `@delay` in the changelog

Reply:
I updated this to `@defer` in [33f5934](https://github.com/apollographql/router/pull/3837/commits/33f593473b6bd3989d3bdb5151c9a7f16a98c4e4)